results/15k_resampled/match_paper_parameters_results.txt
58752/75000
{'task': 'plain_addition', 'decoder': True, 'data_dir': 'data/addition', 'model_dir': 'models', 'results_dir': 'results/15k_resampled', 'resume': False, 'n_embd': 384, 'n_layer': 6, 'n_head': 6, 'dropout': 0.2, 'block_size': 1024, 'batch_size': 12, 'test_batch_size': 2048, 'max_iters': 5000, 'eval_interval': 100, 'min_lr': 0.0001, 'max_lr': 0.001, 'warmup_iters': 100, 'lr_decay_iters': 5000, 'weight_decay': 0.1, 'beta1': 0.9, 'beta2': 0.99, 'seed': 1, 'n_digits': 3, 'n_train': 15000, 'n_val': 10000, 'n_test': 75000, 'use_dollar_signs': False, 'resample_data': True, 'name': 'match_paper_parameters'}
890+989=1889,890+989=1879
799+579=1388,799+579=1378
548+921=1479,548+921=1469
198+978=1186,198+978=1176
459+812=1281,459+812=1271
735+875=1600,735+875=1610
724+416=1130,724+416=1140
529+772=1201,529+772=1301
796+262=1068,796+262=1058
959+718=1687,959+718=1677
547+812=1369,547+812=1359
731+723=1444,731+723=1454
538+612=1140,538+612=1150
984+711=1795,984+711=1695
124+971=1195,124+971=1095
474+606=1070,474+606=1080
894+472=1376,894+472=1366
841+197=1028,841+197=1038
699+368=1077,699+368=1067
496+698=1104,496+698=1194
996+885=1891,996+885=1881
592+987=1589,592+987=1579
158+925=1093,158+925=1083
987+911=1808,987+911=1898
436+784=1210,436+784=1220
582+611=1293,582+611=1193
989+116=1005,989+116=1105
776+352=1118,776+352=1128
694+994=1698,694+994=1688
622+416=1028,622+416=1038
513+889=1302,513+889=1402
538+563=1001,538+563=1101
597+579=1186,597+579=1176
886+615=1401,886+615=1501
696+367=1073,696+367=1063
498+790=1298,498+790=1288
298+780=1088,298+780=1078
671+410=1071,671+410=1081
158+916=1084,158+916=1074
745+735=1470,745+735=1480
973+163=1126,973+163=1136
428+772=1190,428+772=1200
695+654=1359,695+654=1349
676+523=1299,676+523=1199
322+738=1050,322+738=1060
812+286=1198,812+286=1098
996+370=1376,996+370=1366
322+933=1245,322+933=1255
658+541=1299,658+541=1199
697+901=1508,697+901=1598
658+691=1359,658+691=1349
736+999=1725,736+999=1735
192+891=1093,192+891=1083
725+515=1230,725+515=1240
398+776=1184,398+776=1174
233+780=1003,233+780=1013
795+371=1176,795+371=1166
564+636=1100,564+636=1200
893+187=1090,893+187=1080
733+307=1030,733+307=1040
866+614=1470,866+614=1480
999+687=1696,999+687=1686
632+672=1204,632+672=1304
681+359=1030,681+359=1040
732+432=1154,732+432=1164
928+473=1301,928+473=1401
537+733=1260,537+733=1270
803+696=1599,803+696=1499
994+808=1702,994+808=1802
908+971=1889,908+971=1879
875+615=1480,875+615=1490
297+794=1001,297+794=1091
707+994=1601,707+994=1701
810+488=1398,810+488=1298
721+749=1460,721+749=1470
531+569=1090,531+569=1100
630+557=1177,630+557=1187
932+191=1113,932+191=1123
782+408=1180,782+408=1190
325+889=1204,325+889=1214
397+952=1359,397+952=1349
998+925=1933,998+925=1923
914+183=1197,914+183=1097
593+880=1483,593+880=1473
332+828=1150,332+828=1160
599+484=1093,599+484=1083
411+985=1496,411+985=1396
692+482=1184,692+482=1174
944+657=1501,944+657=1601
434+866=1290,434+866=1300
624+477=1001,624+477=1101
966+314=1270,966+314=1280
395+694=1099,395+694=1089
858+591=1459,858+591=1449
975+705=1670,975+705=1680
993+390=1393,993+390=1383
820+482=1202,820+482=1302
925+475=1300,925+475=1400
938+990=1918,938+990=1928
695+771=1476,695+771=1466
991+201=1292,991+201=1192
895+303=1108,895+303=1198
824+836=1650,824+836=1660
445+735=1170,445+735=1180
947+101=1038,947+101=1048
998+185=1193,998+185=1183
498+860=1368,498+860=1358
312+907=1229,312+907=1219
949+710=1669,949+710=1659
998+993=1901,998+993=1991
795+294=1099,795+294=1089
195+909=1004,195+909=1104
598+918=1526,598+918=1516
509+550=1069,509+550=1059
594+781=1385,594+781=1375
331+992=1313,331+992=1323
882+719=1501,882+719=1601
992+107=1009,992+107=1099
994+270=1274,994+270=1264
770+960=1720,770+960=1730
686+510=1296,686+510=1196
688+381=1079,688+381=1069
823+177=990,823+177=1000
695+378=1083,695+378=1073
425+789=1204,425+789=1214
537+903=1430,537+903=1440
752+349=1001,752+349=1101
690+799=1499,690+799=1489
938+412=1340,938+412=1350
889+514=1303,889+514=1403
797+970=1777,797+970=1767
609+991=1500,609+991=1600
329+773=1002,329+773=1102
982+210=1292,982+210=1192
998+291=1299,998+291=1289
597+668=1275,597+668=1265
727+310=1027,727+310=1037
693+608=1201,693+608=1301
234+868=1002,234+868=1102
801+949=1740,801+949=1750
830+408=1228,830+408=1238
742+408=1140,742+408=1150
267+792=1069,267+792=1059
875+252=1117,875+252=1127
698+358=1066,698+358=1056
731+449=1170,731+449=1180
838+190=1018,838+190=1028
718+961=1689,718+961=1679
851+449=1200,851+449=1300
898+500=1308,898+500=1398
791+209=900,791+209=1000
119+960=1089,119+960=1079
995+964=1969,995+964=1959
697+377=1084,697+377=1074
884+813=1797,884+813=1697
995+280=1285,995+280=1275
941+528=1479,941+528=1469
979+880=1869,979+880=1859
194+991=1195,194+991=1185
520+538=1048,520+538=1058
868+916=1794,868+916=1784
621+783=1304,621+783=1404
794+345=1149,794+345=1139
898+280=1188,898+280=1178
997+182=1189,997+182=1179
637+799=1426,637+799=1436
835+765=1590,835+765=1600
431+870=1291,431+870=1301
779+222=901,779+222=1001
797+264=1071,797+264=1061
804+365=1179,804+365=1169
964+138=1002,964+138=1102
832+769=1501,832+769=1601
398+790=1198,398+790=1188
727+531=1248,727+531=1258
621+633=1244,621+633=1254
234+778=1002,234+778=1012
911+487=1498,911+487=1398
622+839=1451,622+839=1461
721+582=1203,721+582=1303
620+730=1340,620+730=1350
796+567=1373,796+567=1363
999+198=1107,999+198=1197
721+389=1100,721+389=1110
997+487=1494,997+487=1484
431+824=1245,431+824=1255
823+231=1044,823+231=1054
797+367=1174,797+367=1164
798+216=1024,798+216=1014
599+540=1149,599+540=1139
597+902=1509,597+902=1499
797+275=1082,797+275=1072
892+926=1828,892+926=1818
498+579=1087,498+579=1077
768+372=1130,768+372=1140
630+871=1491,630+871=1501
844+556=1300,844+556=1400
898+323=1231,898+323=1221
725+685=1400,725+685=1410
981+714=1795,981+714=1695
985+150=1125,985+150=1135
532+570=1002,532+570=1102
537+965=1402,537+965=1502
796+390=1196,796+390=1186
534+920=1444,534+920=1454
920+318=1228,920+318=1238
999+387=1396,999+387=1386
612+689=1201,612+689=1301
490+589=1089,490+589=1079
618+871=1499,618+871=1489
933+213=1136,933+213=1146
798+777=1585,798+777=1575
695+954=1659,695+954=1649
860+290=1140,860+290=1150
206+994=1100,206+994=1200
897+878=1785,897+878=1775
356+941=1397,356+941=1297
993+201=1294,993+201=1194
534+812=1336,534+812=1346
853+747=1500,853+747=1600
223+857=1070,223+857=1080
794+920=1724,794+920=1714
198+877=1085,198+877=1075
465+932=1497,465+932=1397
788+288=1086,788+288=1076
675+541=1206,675+541=1216
614+687=1201,614+687=1301
430+828=1248,430+828=1258
778+323=1001,778+323=1101
903+561=1474,903+561=1464
788+471=1269,788+471=1259
636+409=1035,636+409=1045
306+894=1100,306+894=1200
532+819=1341,532+819=1351
919+940=1869,919+940=1859
723+732=1445,723+732=1455
833+407=1230,833+407=1240
939+309=1238,939+309=1248
698+385=1093,698+385=1083
696+925=1631,696+925=1621
726+632=1348,726+632=1358
563+617=1170,563+617=1180
795+686=1491,795+686=1481
698+389=1097,698+389=1087
874+341=1205,874+341=1215
824+932=1746,824+932=1756
998+578=1586,998+578=1576
899+503=1302,899+503=1402
709+660=1379,709+660=1369
199+916=1125,199+916=1115
632+573=1105,632+573=1205
799+918=1727,799+918=1717
827+475=1202,827+475=1302
894+965=1869,894+965=1859
874+841=1705,874+841=1715
865+285=1140,865+285=1150
723+841=1554,723+841=1564
629+673=1202,629+673=1302
999+277=1286,999+277=1276
589+910=1509,589+910=1499
965+137=1002,965+137=1102
112+983=1195,112+983=1095
633+437=1060,633+437=1070
297+932=1239,297+932=1229
970+726=1796,970+726=1696
916+971=1897,916+971=1887
625+635=1250,625+635=1260
332+838=1160,332+838=1170
538+810=1338,538+810=1348
907+699=1506,907+699=1606
241+909=1140,241+909=1150
943+140=1093,943+140=1083
578+620=1298,578+620=1198
631+402=1023,631+402=1033
821+489=1300,821+489=1310
899+215=1124,899+215=1114
895+369=1274,895+369=1264
221+849=1060,221+849=1070
530+885=1405,530+885=1415
892+757=1659,892+757=1649
833+235=1058,833+235=1068
808+394=1102,808+394=1202
831+986=1807,831+986=1817
984+198=1192,984+198=1182
141+899=1030,141+899=1040
530+487=1007,530+487=1017
537+710=1237,537+710=1247
935+945=1870,935+945=1880
206+894=1000,206+894=1100
594+881=1485,594+881=1475
386+710=1196,386+710=1096
732+678=1400,732+678=1410
595+554=1159,595+554=1149
473+541=1004,473+541=1014
699+600=1209,699+600=1299
527+520=1037,527+520=1047
394+965=1369,394+965=1359
895+380=1285,895+380=1275
534+846=1370,534+846=1380
719+340=1069,719+340=1059
624+939=1553,624+939=1563
903+179=1092,903+179=1082
894+381=1285,894+381=1275
657+444=1001,657+444=1101
794+901=1605,794+901=1695
527+620=1137,527+620=1147
664+362=1016,664+362=1026
909+175=1094,909+175=1084
853+707=1550,853+707=1560
457+842=1399,457+842=1299
833+657=1480,833+657=1490
198+861=1069,198+861=1059
540+763=1203,540+763=1303
895+487=1392,895+487=1382
730+685=1405,730+685=1415
197+874=1081,197+874=1071
831+207=1028,831+207=1038
673+354=1017,673+354=1027
899+162=1071,899+162=1061
520+831=1341,520+831=1351
583+615=1298,583+615=1198
906+175=1091,906+175=1081
897+398=1395,897+398=1295
797+395=1102,797+395=1192
934+229=1153,934+229=1163
533+775=1208,533+775=1308
899+560=1469,899+560=1459
595+587=1192,595+587=1182
897+577=1484,897+577=1474
644+459=1003,644+459=1103
811+281=1192,811+281=1092
930+307=1227,930+307=1237
233+783=1006,233+783=1016
317+784=1001,317+784=1101
916+961=1887,916+961=1877
399+610=1019,399+610=1009
419+870=1299,419+870=1289
794+370=1174,794+370=1164
793+956=1759,793+956=1749
945+655=1500,945+655=1600
835+715=1540,835+715=1550
906+899=1705,906+899=1805
497+996=1403,497+996=1493
994+117=1011,994+117=1111
359+940=1309,359+940=1299
296+803=1199,296+803=1099
291+928=1229,291+928=1219
691+381=1082,691+381=1072
608+995=1503,608+995=1603
728+309=1027,728+309=1037
897+234=1141,897+234=1131
919+278=1297,919+278=1197
908+365=1283,908+365=1273
715+485=1100,715+485=1200
299+782=1091,299+782=1081
594+545=1149,594+545=1139
990+413=1303,990+413=1403
534+566=1000,534+566=1100
223+831=1044,223+831=1054
801+598=1499,801+598=1399
895+878=1783,895+878=1773
766+435=1101,766+435=1201
699+967=1676,699+967=1666
858+925=1793,858+925=1783
865+315=1170,865+315=1180
826+844=1660,826+844=1670
434+606=1030,434+606=1040
352+767=1129,352+767=1119
720+291=1001,720+291=1011
830+413=1233,830+413=1243
904+261=1175,904+261=1165
697+990=1697,697+990=1687
112+999=1101,112+999=1111
833+219=1042,833+219=1052
255+944=1299,255+944=1199
387+811=1208,387+811=1198
699+974=1683,699+974=1673
681+606=1277,681+606=1287
531+635=1156,531+635=1166
766+237=903,766+237=1003
866+314=1170,866+314=1180
332+771=1003,332+771=1103
524+536=1050,524+536=1060
294+995=1299,294+995=1289
998+365=1373,998+365=1363
936+194=1120,936+194=1130
594+707=1201,594+707=1301
490+899=1399,490+899=1389
965+335=1200,965+335=1300
697+495=1102,697+495=1192
753+607=1350,753+607=1360
699+484=1193,699+484=1183
795+571=1376,795+571=1366
630+673=1203,630+673=1303
874+353=1217,874+353=1227
132+986=1108,132+986=1118
543+807=1340,543+807=1350
995+790=1795,995+790=1785
595+874=1479,595+874=1469
598+579=1187,598+579=1177
837+664=1401,837+664=1501
987+120=1007,987+120=1107
638+400=1028,638+400=1038
727+321=1038,727+321=1048
476+804=1270,476+804=1280
631+614=1235,631+614=1245
499+687=1196,499+687=1186
534+869=1303,534+869=1403
608+591=1209,608+591=1199
898+304=1212,898+304=1202
634+716=1340,634+716=1350
823+735=1548,823+735=1558
437+964=1301,437+964=1401
258+917=1185,258+917=1175
458+701=1169,458+701=1159
898+490=1398,898+490=1388
358+827=1195,358+827=1185
837+203=1030,837+203=1040
795+971=1776,795+971=1766
793+372=1175,793+372=1165
937+443=1370,937+443=1380
796+275=1081,796+275=1071
958+443=1301,958+443=1401
196+990=1196,196+990=1186
867+434=1201,867+434=1301
798+316=1124,798+316=1114
317+883=1100,317+883=1200
195+981=1186,195+981=1176
381+969=1340,381+969=1350
901+468=1379,901+468=1369
330+675=905,330+675=1005
396+904=1200,396+904=1300
894+894=1798,894+894=1788
873+807=1670,873+807=1680
698+667=1375,698+667=1365
973+627=1500,973+627=1600
697+577=1284,697+577=1274
797+499=1396,797+499=1296
933+847=1770,933+847=1780
479+711=1290,479+711=1190
769+232=901,769+232=1001
232+786=1008,232+786=1018
758+411=1179,758+411=1169
512+983=1595,512+983=1495
897+297=1104,897+297=1194
986+518=1404,986+518=1504
940+919=1869,940+919=1859
506+899=1305,506+899=1405
695+928=1633,695+928=1623
545+655=1100,545+655=1200
746+456=1102,746+456=1202
530+876=1396,530+876=1406
926+354=1270,926+354=1280
821+739=1550,821+739=1560
910+780=1790,910+780=1690
722+932=1644,722+932=1654
516+981=1597,516+981=1497
732+630=1352,732+630=1362
934+569=1403,934+569=1503
694+855=1559,694+855=1549
894+490=1394,894+490=1384
545+735=1270,545+735=1280
524+931=1445,524+931=1455
808+961=1779,808+961=1769
892+487=1389,892+487=1379
430+632=1052,430+632=1062
959+598=1567,959+598=1557
912+270=1192,912+270=1182
222+988=1200,222+988=1210
695+889=1594,695+889=1584
889+808=1797,889+808=1697
984+354=1328,984+354=1338
893+610=1403,893+610=1503
910+786=1796,910+786=1696
794+489=1293,794+489=1283
613+587=1100,613+587=1200
806+196=902,806+196=1002
531+629=1150,531+629=1160
664+336=900,664+336=1000
630+918=1538,630+918=1548
896+293=1199,896+293=1189
990+919=1019,990+919=1909
799+760=1569,799+760=1559
733+868=1501,733+868=1601
874+326=1100,874+326=1200
793+271=1074,793+271=1064
894+581=1485,894+581=1475
873+327=1100,873+327=1200
531+615=1136,531+615=1146
838+902=1730,838+902=1740
331+784=1105,331+784=1115
422+828=1240,422+828=1250
988+115=1003,988+115=1103
706+390=1196,706+390=1096
823+732=1545,823+732=1555
794+789=1593,794+789=1583
675+510=1175,675+510=1185
532+833=1355,532+833=1365
554+747=1201,554+747=1301
796+360=1166,796+360=1156
281+818=1199,281+818=1099
917+280=1297,917+280=1197
972+429=1301,972+429=1401
897+188=1095,897+188=1085
837+403=1230,837+403=1240
915+167=1092,915+167=1082
524+877=1301,524+877=1401
708+651=1369,708+651=1359
524+716=1230,524+716=1240
255+941=1296,255+941=1196
534+612=1136,534+612=1146
324+899=1213,324+899=1223
702+394=1196,702+394=1096
920+260=1170,920+260=1180
899+476=1385,899+476=1375
599+465=1074,599+465=1064
697+568=1275,697+568=1265
192+923=1125,192+923=1115
596+703=1209,596+703=1299
959+856=1825,959+856=1815
699+301=900,699+301=1000
834+366=1100,834+366=1200
895+479=1384,895+479=1374
608+892=1400,608+892=1500
298+887=1195,298+887=1185
827+193=1010,827+193=1020
903+698=1501,903+698=1601
358+656=1024,358+656=1014
775+427=1102,775+427=1202
334+746=1070,334+746=1080
897+504=1301,897+504=1401
727+923=1640,727+923=1650
631+413=1034,631+413=1044
698+523=1231,698+523=1221
832+622=1444,832+622=1454
531+707=1228,531+707=1238
223+988=1201,223+988=1211
960+999=1969,960+999=1959
336+866=1102,336+866=1202
898+476=1384,898+476=1374
905+368=1283,905+368=1273
685+512=1297,685+512=1197
634+606=1230,634+606=1240
898+421=1329,898+421=1319
222+880=1002,222+880=1102
511+983=1594,511+983=1494
652+708=1350,652+708=1360
694+390=1094,694+390=1084
331+732=1053,331+732=1063
730+418=1138,730+418=1148
638+899=1527,638+899=1537
588+881=1479,588+881=1469
692+491=1193,692+491=1183
695+878=1583,695+878=1573
795+904=1709,795+904=1699
403+696=1199,403+696=1099
420+584=904,420+584=1004
817+671=1498,817+671=1488
498+800=1208,498+800=1298
920+179=1199,920+179=1099
372+708=1070,372+708=1080
359+656=1025,359+656=1015
609+970=1589,609+970=1579
839+979=1808,839+979=1818
974+140=1104,974+140=1114
793+469=1272,793+469=1262
638+782=1410,638+782=1420
806+894=1600,806+894=1700
897+272=1179,897+272=1169
583+616=1299,583+616=1199
995+268=1273,995+268=1263
326+977=1203,326+977=1303
939+666=1505,939+666=1605
230+991=1211,230+991=1221
797+281=1088,797+281=1078
999+677=1686,999+677=1676
131+881=1002,131+881=1012
423+833=1246,423+833=1256
795+287=1092,795+287=1082
116+973=1099,116+973=1089
599+467=1076,599+467=1066
796+581=1387,796+581=1377
879+159=1028,879+159=1038
981+162=1133,981+162=1143
892+123=1025,892+123=1015
938+789=1717,938+789=1727
276+824=1000,276+824=1100
999+302=1201,999+302=1301
958+596=1564,958+596=1554
832+758=1580,832+758=1590
799+674=1483,799+674=1473
730+486=1206,730+486=1216
277+803=1070,277+803=1080
398+890=1298,398+890=1288
631+633=1254,631+633=1264
874+352=1216,874+352=1226
992+382=1384,992+382=1374
833+327=1150,833+327=1160
407+896=1203,407+896=1303
827+776=1503,827+776=1603
890+589=1489,890+589=1479
727+575=1202,727+575=1302
935+766=1601,935+766=1701
831+233=1054,831+233=1064
633+847=1470,633+847=1480
723+317=1030,723+317=1040
725+745=1460,725+745=1470
534+879=1403,534+879=1413
766+434=1100,766+434=1200
996+880=1886,996+880=1876
228+873=1001,228+873=1101
896+982=1888,896+982=1878
880+818=1798,880+818=1698
473+807=1270,473+807=1280
757+943=1600,757+943=1700
218+901=1129,218+901=1119
863+437=1200,863+437=1300
451+668=1129,451+668=1119
698+400=1108,698+400=1098
332+984=1306,332+984=1316
337+753=1080,337+753=1090
797+801=1508,797+801=1598
595+574=1179,595+574=1169
220+995=1205,220+995=1215
938+109=1037,938+109=1047
796+700=1406,796+700=1496
488+781=1279,488+781=1269
764+939=1603,764+939=1703
731+949=1670,731+949=1680
929+899=1818,929+899=1828
399+820=1229,399+820=1219
832+579=1401,832+579=1411
775+605=1370,775+605=1380
343+979=1312,343+979=1322
943+559=1402,943+559=1502
922+708=1620,922+708=1630
468+631=1199,468+631=1099
574+806=1370,574+806=1380
997+507=1404,997+507=1504
694+388=1092,694+388=1082
885+615=1400,885+615=1500
827+753=1570,827+753=1580
729+308=1027,729+308=1037
327+874=1101,327+874=1201
335+975=1300,335+975=1310
821+933=1744,821+933=1754
836+544=1370,836+544=1380
997+784=1791,997+784=1781
531+919=1440,531+919=1450
998+501=1409,998+501=1499
693+882=1585,693+882=1575
569+627=1296,569+627=1196
698+992=1600,698+992=1690
487+710=1297,487+710=1197
670+517=1177,670+517=1187
231+889=1110,231+889=1120
331+987=1308,331+987=1318
638+362=900,638+362=1000
794+481=1285,794+481=1275
894+575=1479,894+575=1469
223+899=1112,223+899=1122
604+860=1474,604+860=1464
926+532=1448,926+532=1458
565+537=1002,565+537=1102
890+123=1023,890+123=1013
976+126=1002,976+126=1102
730+708=1428,730+708=1438
638+963=1501,638+963=1601
433+857=1280,433+857=1290
896+787=1693,896+787=1683
826+544=1360,826+544=1370
435+879=1304,435+879=1314
725+729=1444,725+729=1454
895+780=1685,895+780=1675
790+298=1098,790+298=1088
597+895=1402,597+895=1492
718+771=1499,718+771=1489
736+664=1390,736+664=1400
629+989=1608,629+989=1618
298+918=1226,298+918=1216
733+931=1654,733+931=1664
497+692=1199,497+692=1189
535+875=1400,535+875=1410
655+625=1270,655+625=1280
686+513=1299,686+513=1199
293+907=1100,293+907=1200
633+413=1036,633+413=1046
833+747=1570,833+747=1580
882+137=1029,882+137=1019
678+323=901,678+323=1001
324+779=1003,324+779=1103
993+380=1383,993+380=1373
241+928=1179,241+928=1169
998+512=1410,998+512=1510
501+599=1000,501+599=1100
894+374=1278,894+374=1268
820+190=1000,820+190=1010
775+635=1400,775+635=1410
418+661=1089,418+661=1079
831+433=1254,831+433=1264
627+673=1290,627+673=1300
992+697=1799,992+697=1689
692+582=1284,692+582=1274
698+884=1592,698+884=1582
792+357=1159,792+357=1149
872+128=990,872+128=1000
992+287=1289,992+287=1279
969+814=1793,969+814=1783
792+387=1189,792+387=1179
132+979=1101,132+979=1111
898+399=1397,898+399=1297
733+429=1152,733+429=1162
870+620=1480,870+620=1490
299+914=1223,299+914=1213
850+187=1027,850+187=1037
599+977=1586,599+977=1576
897+317=1224,897+317=1214
433+887=1310,433+887=1320
899+464=1373,899+464=1363
622+668=1280,622+668=1290
809+792=1501,809+792=1601
730+686=1406,730+686=1416
690+489=1189,690+489=1179
822+635=1447,822+635=1457
435+875=1300,435+875=1310
654+349=903,654+349=1003
358+841=1209,358+841=1199
999+767=1776,999+767=1766
827+574=1301,827+574=1401
646+456=1002,646+456=1102
847+454=1201,847+454=1301
894+898=1702,894+898=1792
672+518=1180,672+518=1190
895+296=1101,895+296=1191
440+880=1310,440+880=1320
675+326=901,675+326=1001
496+918=1424,496+918=1414
798+686=1494,798+686=1484
185+915=1000,185+915=1100
798+556=1364,798+556=1354
795+570=1375,795+570=1365
693+482=1185,693+482=1175
305+996=1201,305+996=1301
496+676=1182,496+676=1172
930+687=1607,930+687=1617
198+925=1133,198+925=1123
877+363=1230,877+363=1240
588+795=1393,588+795=1383
998+106=1004,998+106=1104
798+773=1581,798+773=1571
332+788=1110,332+788=1120
693+401=1194,693+401=1094
257+857=1124,257+857=1114
984+211=1295,984+211=1195
896+678=1584,896+678=1574
697+522=1229,697+522=1219
678+481=1169,678+481=1159
897+402=1209,897+402=1299
738+979=1707,738+979=1717
987+615=1502,987+615=1602
417+972=1399,417+972=1389
297+974=1281,297+974=1271
773+272=1035,773+272=1045
837+420=1247,837+420=1257
922+728=1640,922+728=1650
459+912=1381,459+912=1371
695+356=1061,695+356=1051
326+874=1190,326+874=1200
797+798=1505,797+798=1595
735+316=1041,735+316=1051
997+891=1898,997+891=1888
700+999=1799,700+999=1699
598+593=1101,598+593=1191
627+719=1336,627+719=1346
895+294=1199,895+294=1189
661+520=1171,661+520=1181
752+308=1050,752+308=1060
892+271=1173,892+271=1163
198+909=1007,198+909=1107
673+341=1004,673+341=1014
859+813=1682,859+813=1672
935+193=1118,935+193=1128
564+537=1001,564+537=1101
492+608=1000,492+608=1100
998+780=1788,998+780=1778
794+768=1572,794+768=1562
141+876=1007,141+876=1017
496+588=1094,496+588=1084
624+726=1340,624+726=1350
996+910=1806,996+910=1906
892+206=1008,892+206=1098
595+880=1485,595+880=1475
520+535=1045,520+535=1055
900+603=1403,900+603=1503
678+811=1499,678+811=1489
809+992=1701,809+992=1801
931+391=1312,931+391=1322
930+393=1313,930+393=1323
996+378=1384,996+378=1374
731+302=1023,731+302=1033
916+681=1507,916+681=1597
998+196=1104,998+196=1194
822+308=1120,822+308=1130
893+907=1700,893+907=1800
911+288=1299,911+288=1199
320+880=1190,320+880=1200
660+920=1570,660+920=1580
918+761=1689,918+761=1679
623+999=1612,623+999=1622
696+395=1001,696+395=1091
117+972=1099,117+972=1089
333+788=1111,333+788=1121
796+601=1307,796+601=1397
795+654=1459,795+654=1449
884+811=1795,884+811=1695
595+705=1200,595+705=1300
995+298=1203,995+298=1293
433+647=1070,433+647=1080
855+944=1899,855+944=1799
651+950=1501,651+950=1601
632+634=1256,632+634=1266
597+457=1064,597+457=1054
997+691=1698,997+691=1688
698+631=1339,698+631=1329
811+939=1740,811+939=1750
630+750=1370,630+750=1380
937+669=1506,937+669=1606
807+390=1297,807+390=1197
730+406=1126,730+406=1136
819+276=1195,819+276=1095
594+995=1599,594+995=1589
997+588=1595,997+588=1585
747+553=1200,747+553=1300
851+189=1030,851+189=1040
923+132=1045,923+132=1055
624+780=1304,624+780=1404
717+862=1589,717+862=1579
592+467=1069,592+467=1059
736+524=1250,736+524=1260
892+117=1019,892+117=1009
895+571=1476,895+571=1466
703+898=1501,703+898=1601
692+983=1685,692+983=1675
110+987=1197,110+987=1097
320+984=1294,320+984=1304
889+900=1799,889+900=1789
525+577=1002,525+577=1102
898+860=1768,898+860=1758
699+662=1371,699+662=1361
844+656=1400,844+656=1500
999+270=1279,999+270=1269
850+453=1203,850+453=1303
623+588=1201,623+588=1211
939+889=1818,939+889=1828
638+420=1048,638+420=1058
811+190=901,811+190=1001
939+664=1503,939+664=1603
278+791=1079,278+791=1069
468+730=1298,468+730=1198
133+967=1090,133+967=1100
909+401=1300,909+401=1310
694+695=1399,694+695=1389
713+970=1693,713+970=1683
418+970=1398,418+970=1388
697+778=1485,697+778=1475
907+994=1801,907+994=1901
796+623=1429,796+623=1419
607+996=1503,607+996=1603
972+129=1001,972+129=1101
732+332=1054,732+332=1064
987+813=1700,987+813=1800
608+951=1569,608+951=1559
387+812=1299,387+812=1199
762+438=1190,762+438=1200
455+644=1199,455+644=1099
823+431=1244,823+431=1254
940+111=1041,940+111=1051
698+974=1682,698+974=1672
836+534=1360,836+534=1370
909+492=1301,909+492=1401
987+706=1793,987+706=1693
597+524=1131,597+524=1121
772+514=1276,772+514=1286
485+913=1498,485+913=1398
867+992=1869,867+992=1859
792+627=1429,792+627=1419
886+140=1016,886+140=1026
799+791=1500,799+791=1590
829+291=1110,829+291=1120
996+161=1167,996+161=1157
899+904=1813,899+904=1803
895+386=1291,895+386=1281
923+832=1745,923+832=1755
977+195=1182,977+195=1172
830+586=1406,830+586=1416
735+445=1170,735+445=1180
798+769=1577,798+769=1567
324+886=1200,324+886=1210
188+891=1089,188+891=1079
584+715=1399,584+715=1299
898+696=1504,898+696=1594
721+378=1199,721+378=1099
994+582=1586,994+582=1576
793+571=1374,793+571=1364
526+579=1005,526+579=1105
895+974=1879,895+974=1869
845+959=1704,845+959=1804
871+929=1700,871+929=1800
623+538=1151,623+538=1161
886+811=1797,886+811=1697
820+847=1657,820+847=1667
196+981=1187,196+981=1177
597+978=1585,597+978=1575
977+623=1500,977+623=1600
724+439=1153,724+439=1163
995+987=1992,995+987=1982
899+363=1272,899+363=1262
612+790=1302,612+790=1402
590+892=1492,590+892=1482
899+973=1882,899+973=1872
962+228=1180,962+228=1190
599+582=1191,599+582=1181
735+525=1250,735+525=1260
495+797=1202,495+797=1292
688+997=1695,688+997=1685
620+940=1550,620+940=1560
433+627=1050,433+627=1060
949+720=1679,949+720=1669
931+222=1143,931+222=1153
458+711=1179,458+711=1169
187+914=1001,187+914=1101
798+785=1593,798+785=1583
995+296=1201,995+296=1291
664+506=1160,664+506=1170
621+719=1330,621+719=1340
696+772=1478,696+772=1468
591+509=1090,591+509=1100
496+782=1288,496+782=1278
509+750=1269,509+750=1259
692+590=1292,692+590=1282
994+519=1413,994+519=1513
931+515=1436,931+515=1446
870+320=1180,870+320=1190
731+999=1720,731+999=1730
636+990=1616,636+990=1626
902+700=1502,902+700=1602
957+815=1782,957+815=1772
773+827=1500,773+827=1600
870+170=1030,870+170=1040
966+539=1405,966+539=1505
793+578=1381,793+578=1371
692+591=1293,692+591=1283
783+616=1499,783+616=1399
997+915=1922,997+915=1912
771+289=1050,771+289=1060
113+976=1099,113+976=1089
196+913=1119,196+913=1109
537+899=1426,537+899=1436
694+988=1692,694+988=1682
540+960=1400,540+960=1500

util/config.py
import json
from typing import Literal


class Config:
    """Configuration for data generation, training, and evaluation."""

    batch_size: int = 64
    beta1: float = 0.9
    beta2: float = 0.99
    block_size: int = 64
    data_dir: str = "data/addition"
    decoder: bool = True
    dropout: float = 0.0
    eval_interval: int = 100
    lr_decay_iters: int = 5000
    max_evals_without_improving: int = 10
    max_iters: int = 5000
    max_loss_for_early_stopping: float = 1.1
    max_lr: float = 5e-4
    min_lr: float = 5e-5
    model_dir: str = "models"
    n_digits: int = 3
    n_embd: int = 384
    n_head: int = 6
    n_layer: int = 6
    n_test: int = 75000
    n_train: int = 10000
    n_val: int = 10000
    name: str = ""
    resample_data: bool = True
    results_dir: str = "results"
    resume: bool = False
    seed: int = 42
    task: Literal[
        "plain_addition", "reversed_addition", "shakespeare"
    ] = "plain_addition"
    test_batch_size: int = 2048
    use_delimiter: bool = True
    warmup_iters: int = 100
    weight_decay: float = 0.1

    def __init__(self, config: dict[str, bool | int | float | str]) -> None:
        for k, v in config.items():
            setattr(self, k, v)

        assert self.name != ""

    def to_dict(self) -> dict[str, bool | int | float | str]:
        """Returns a `dict` with all configuration information."""
        d = {}
        for k in Config.__dict__.keys():
            if "__" not in k and k not in ("from_json", "to_dict"):
                d[k] = getattr(self, k)

        return d

    @property
    def checkpoint_name(self) -> str:
        return self.name + ".pt"

    @staticmethod
    def from_json(path: str) -> "Config":
        with open(path, "r") as f:
            config = json.load(f)

        return Config(config)
